x/chat/types/channel.go

```diff
@@ -0,0 +1,52 @@
+package types
+
+import (
+	types "github.com/cosmos/cosmos-sdk/codec/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const (
+	ChannelNameMaxLength    = 50   // TODO: Decide this value or make it customizable through params
+	ChannelSubjectMaxLength = 1000 // TODO: Decide this value or make it customizable through params
+)
+
+// NewChannel creates a new channel
+func NewChannel(
+	id int32,
+	creator User,
+	name string,
+	subject string,
+	metadata *types.Any,
+) (Channel, error) {
+	channel := new(Channel)
```

Comment:
to not dereference on returns, maybe we can initialize these like this

```suggestion
	channel := Channel{}
```

Reply:
Thanks for the comment, I used `new` because the function was returning pointer at first but it actually complicates the code for nothing.
I'm not sure you can have dereference on return in Go though